### DISCUSSION TITLE
multistage build vs compose override

### DISCUSSION THREAD

**Discussion by u2ros** [BretFisher/ama#144] [Q&A]
Hi,

I'm going back over your node and docker course and I see that before multi-stage builds are introduced, the approach you use with docker-compose for local dev is, that you override certain aspects, for example you override the command stanza to use nodemon and you set the NODE_ENV in the compose yaml file. Later, when multistage builds are introduced you just use the target stanza in the compose file and use a specific build stage from Dockerfile.

Would you consider the first approach kind of old way of doing things (maybe you did it that way for the flow of the course) and we should kind of take advantage of multistage build approach nowadays?

**Comment by BretFisher** [comment]
Both are fine, and it depends on your goals.

If the whole team wants to use a specific config and you'd rather have it as a stage in the Dockerfile, cool. But if you're sort of a one-off developer config, then it makes more sense to put in your own `docker-compose.override.yml` that will automatically get picked up on your machine (and should be in `.gitignore`).

Either way is the same result and there's really no benefit in speed or image size, it's just a team/personal preference IMO.
